pkg/server/container_stats.go

```diff
@@ -32,19 +32,22 @@ func (c *criContainerdService) ContainerStats(ctx context.Context, in *runtime.C
 		return nil, fmt.Errorf("invalid container stats request")
 	}
 	containerID := in.GetContainerId()
-	_, err := c.containerStore.Get(containerID)
+	cntr, err := c.containerStore.Get(containerID)
 	if err != nil {
-		return nil, fmt.Errorf("failed to find container %q: %v", containerID, err)
+		return nil, fmt.Errorf("failed to find container: %v", err)
 	}
 	request := &tasks.MetricsRequest{Filters: []string{"id==" + containerID}}
 	resp, err := c.taskService.Metrics(ctx, request)
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch metrics for tasks: %v", err)
+		return nil, fmt.Errorf("failed to fetch metrics for task: %v", err)
+	}
+	if len(resp.Metrics) != 1 {
+		return nil, fmt.Errorf("unexpected metrics response: %+v", resp.Metrics)
 	}
 
-	var cs runtime.ContainerStats
-	if err := c.getContainerMetrics(containerID, resp.Metrics[0], &cs); err != nil {
-		return nil, fmt.Errorf("failed to decode container metrics: %v", err)
+	cs, err := c.getContainerMetrics(cntr.Metadata, resp.Metrics[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate container metrics: %v", err)
```

Comment:
I would stick with decode. We are not technically generating :)

Reply:
Will do.